corehq/apps/userreports/sql/adapter.py

```diff
@@ -216,6 +228,30 @@ def doc_exists(self, doc):
             query = session.query(self.get_table()).filter_by(doc_id=doc['_id'])
             return session.query(query.exists()).scalar()
 
+    def get_docs(self, doc_id):
+        """Returns a list of entries from the datasource matching the `doc_id`. Each entry is a dictionary with
+        the column name as the key and value as the value
+        """
+        table = self.get_table()
+        with self.session_context() as session:
+            query = session.query(self.get_table()).filter_by(doc_id=doc_id)
+
+        def get_table(query):
+            yield list(table.columns.keys())
+            for row in query:
+                yield row
+
+        table_ = get_table(query)
+        headers = next(table_)
```

Comment:
you could just remove the first yield from `get_table` and use it directly here:
```python
headers = list(table.columns.keys())
``` 

I think that makes `get_table` redundant

Reply:
https://github.com/dimagi/commcare-hq/pull/33877/commits/e0c5698e49eb7d6fc68053757a533582e245926d